policies/committer-policy.md

```diff
@@ -0,0 +1,96 @@
+
+# Policy for OpenSSL Committers
+
+## Who is a committer?
+
+OpenSSL committers are contributors who have commit access to the OpenSSL
+source code repository.  Committers review and commit their own patches
+as well as those of other contributors.
+
+## How to become a committer?
+
+Commit access is granted by the [OMC] typically on the recommendation
+of the [OTC] (see the OpenSSL Bylaws).
+
+We welcome contributors who become domain experts in some part of
+the library (for example, low-level crypto) as well as generalists
+who contribute to all areas of the codebase.  All committers share
+the responsibility for the overall health of the project: aside from
+contributing quality features, committers are team players who fix bugs,
+address open issues, review community contributions, and improve tests
+and documentation.  Committers are also shepherds of the OpenSSL community
+and its code of conduct.
+
+To become a committer, start by contributing code.  Read our coding style,
+and get to know our build and test system.  Then, use the Github issue
+tracker, and our mailing lists find impactful ideas to work on.
+
+## How to maintain committer status?
+
+To maintain committer status, you must stay active in the project.  Refer
+to the OpenSSL Bylaws for details.
+
+In the unlikely and unfortunate event that your actions conflict with
+the project objectives or are otherwise disruptive, committer status
+may also be revoked by the OMC.
+
+## Approvals and code reviews
+
+All submissions must be reviewed and approved by at least two committers,
+one of whom must also be an OTC member.  If the author is also a committer
+then that counts as one of the reviews.  In other words:
+
+* OTC members need one approval from any committer
+* Committers need one approval from an OTC member
+* Contributors without commit rights need two approvals, including
```

Comment:
I'm having some concerns about author's reviews not counting for them selves.

* releases require commits to be reviewed -- does this mean three people for each release instead of two?  If so, who?  We have trouble finding two at times.  These commits are mechanical and a second set of eyes seems reasonable.
* If a third party submits a PR and one of us adds a test case, does this make them unable to review any of the submission?  The submission would require two additional reviews for the test case but the entire thing?
* If two of us cooperate to create a PR (which happens more than occasionally), does this require two new reviewers?

There are four staff who can currently sign off reviews and by far the bulk of reviewing is done by these four.  Insisting all four sign off seems problematic, even three could be difficult.

Reply:
I think we can put in place a policy that handles the mechanical items needing during a release as a separate context where basically the two people involved in the release are the reviewers - as basically it is mechanical stuff rather than creative and in need of dual review.

For test case addition - they can always be placed in a separate PR if you are concerned about review challenges.

And as per previous discussions, we can make a policy change like this and see how it works out in practice and adjust after we have more information. Not changing a policy because we are concerned about something that isn't certain to happen is concerning.

And of course the number of paid resources is increasing so the "four staff" issue should be less of a concern over time.

Reply:
Paid resources don't being as either committer or OTC.  Thus, they don't have any official review capability.  At this stage, we don't know how long it takes.  Given the usual rate of progress, months seems likely.

I have worded out an exception for the release.  Being mechanical, there isn't much to review really.

I was struggling with the myriad other cases as exceptions.  I'll keep trying.  Personally, I don't have a problem with one submitter reviewing bits done by another so long as there is a third sign off.  Codifying this nicely, on the other hand, isn't so easy :)

Reply:
If we are going to be talking about the author's review *not* counting as one of the two reviews, should the text in the PR be modified to reflect that?  I currently see

> If the author is also a committer then that counts as one of the reviews

which does not seem consistent with the ongoing discussion here.

Reply:
This is the original text slightly updated.

I'm torn between getting the current policy in and then updating it or trying to do the update as part of this PR.

Reply:
How about this wording:

> All submissions must be reviewed and approved by at least two committers,
> one of whom must also be an OTC member.  Neither of the reviewers can
> be the author of the submission.
> 
> The sole exception to this is for the release procedure commits where
> the author's review does count towards the two needed.
> 
> In the case where two committers make a joint submission, they can review
> each other's code but not their own.  A third reviewer will be required.

To replace the text in the paragraph and the bullet points.

Reply:
That looks like a reasonable and pragmatic description to match the intent of the change.
Updating the policy is the path you've started down rather than just reformatting so I'd stick to that now.

Reply:
Okay.  I'll still add this as a separate commit in case we need to revert back to what was there.